zerver/tests/test_message_edit.py

```diff
@@ -4829,6 +4829,97 @@ def test_delete_message_by_other_user(msg_id: int) -> "TestHttpResponse":
             result = test_delete_message_by_owner(msg_id=msg_id)
             self.assert_json_error(result, "Message already deleted")
 
+    def test_delete_message_sent_by_bots(self) -> None:
+        def set_message_deleting_params(
+            delete_own_message_policy: int, message_content_delete_limit_seconds: Union[int, str]
+        ) -> None:
+            self.login("iago")
+            result = self.client_patch(
```

Comment:
Usually it's cleaner to use `api_patch` for this sort of thing -- changing which user you're logged in as in a helper function can create land mines, where you write a test that logs in to a user and then call a helper function, and don't realize the helper changed who you're logged in as.